File: app/src/main/java/com/example/mobileproject/Adapter/FirestoreRecyclerAdapter.java
package com.example.mobileproject.Adapter;

import android.support.annotation.NonNull;
import android.view.ViewGroup;

import com.example.mobileproject.Activity.PostActivity;
import com.example.mobileproject.holder.HomeItemHolder;
import com.example.mobileproject.model.DetailItem;
import com.firebase.ui.firestore.FirestoreRecyclerOptions;

public abstract class FirestoreRecyclerAdapter extends com.firebase.ui.firestore.FirestoreRecyclerAdapter<DetailItem, HomeItemHolder> {
    /**
     * Create a new RecyclerView adapter that listens to a Firestore Query.  See {@link
     * FirestoreRecyclerOptions} for configuration options.
     *
     * @param options
     */
    public FirestoreRecyclerAdapter(@NonNull FirestoreRecyclerOptions<DetailItem> options) {
        super(options);

    }

    @Override
    protected abstract void onBindViewHolder(@NonNull HomeItemHolder holder, int position, DetailItem model);

    @NonNull
    @Override
    public abstract HomeItemHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i);
}

File: app/src/main/java/com/example/mobileproject/Adapter/FirestoreRecyclerAdapter.kt
package com.example.mobileproject.Adapter

import android.view.ViewGroup
import com.example.mobileproject.holder.HomeItemHolder
import com.example.mobileproject.model.DetailItem
import com.firebase.ui.firestore.FirestoreRecyclerOptions

abstract class FirestoreRecyclerAdapter
/**
 * Create a new RecyclerView adapter that listens to a Firestore Query.  See [ ] for configuration options.
 *
 * @param options
 */
(options: FirestoreRecyclerOptions<DetailItem>) : com.firebase.ui.firestore.FirestoreRecyclerAdapter<DetailItem, HomeItemHolder>(options) {

    abstract override fun onBindViewHolder(holder: HomeItemHolder, position: Int, model: DetailItem)

    abstract override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): HomeItemHolder
}